src/Model/IndexInfoIteratorIterator.php

```diff
@@ -34,11 +34,13 @@
  * @see https://github.com/mongodb/specifications/blob/master/source/enumerate-indexes.rst
  * @see https://mongodb.com/docs/manual/reference/command/listIndexes/
  * @see https://mongodb.com/docs/manual/reference/system-collections/
+ * @template-extends IteratorIterator<int, array, Traversable<int, array>>
```

Comment:
Why are there three types under `IteratorIterator<>`? Are the first two types always consistent with the two types for Traversable? Presumably "Traversable" here is just whatever is returned by `getIterator()`?

Reply:
The template for `IteratorIterator` includes the inner iterator, unfortunately also locking it to the same key type: https://github.com/vimeo/psalm/blob/9c814c8a69b0fd80f7fb81de5740434ed1ce7022/stubs/CoreGenericIterators.phpstub#L105..L111. This has already proved problematic in the `AsListIterator` I was working on for our Codec implementation, so I'll have to revisit this to figure out what the purpose of the third template parameter is. This works for us, so we're good.